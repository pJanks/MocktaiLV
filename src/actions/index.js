export const addUser = (user) => ({
  type: 'ADD_USER',
  user
})

export const addDrinks = (drinks) => ({
  type: 'LOAD_NONALCOHOLIC_DRINKS',
  drinks
})

export const addFavoriteDrink = (favoriteDrink) => ({
  type: 'ADD_FAVORITE_DRINK',
  favoriteDrink
})

export const getDrink = (drink) => ({
  type: 'GET_DRINK',
  drink
})
