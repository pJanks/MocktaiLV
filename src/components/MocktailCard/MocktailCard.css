article {
  border: 1px solid #000;
  display: flex;
  flex-direction: column;
  margin: 20px;
  padding: 13px;
}
