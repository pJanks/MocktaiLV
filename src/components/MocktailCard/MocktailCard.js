import React, { Component } from 'react';
import DrinkDetails from '../DrinkDetails/DrinkDetails'
import PropTypes from 'prop-types'
import { Link, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { addFavoriteDrink } from '../../actions'
import { getDrink } from '../../actions'
import './MocktailCard.css';

let favoriteDrinks = []


export class MocktailCard extends Component {
  constructor() {
    super()
  }

handleFavoriteButtonClick = (drink) => {
    if (!this.props.user.userVerified) {
      window.alert('You must be logged in to do that');
    } else {
      favoriteDrinks.push(drink)
      this.props.addToUserFavorites(this.props.drink);
      localStorage.setItem(this.props.user.name, JSON.stringify({
        name: this.props.user.name,
        password: this.props.user.password,
        favorites: favoriteDrinks
      }))
    }
  }

  handleDetailsButtonClick = (drink) => {
    this.props.addDrinkToStore(drink)
  }


render() {
  return(
    <article>
      <h3>{this.props.drink.strDrink}</h3>
      <button onClick={() => this.handleFavoriteButtonClick(this.props.drink)}>Favorite</button>
      <button onClick={() => this.handleDetailsButtonClick(this.props.drink)} ><Link className='link' to={`/DrinkDetails${this.props.drink.idDrink}`}>Details</Link></button>
      <img src={this.props.drink.strDrinkThumb} alt='drink image' className='drink-image'/>
    </article>
  )}
}


export const mapDispatchToProps = (dispatch) => ({
  addToUserFavorites: (favoriteDrink) => { dispatch(addFavoriteDrink(favoriteDrink)) },
  addDrinkToStore: (drink) => { dispatch(getDrink(drink)) }
})

export const mapStateToProps = (state) => ({
  user: state.user
})

MocktailCard.propTypes = {
  user: PropTypes.object
}

export default connect(mapStateToProps, mapDispatchToProps)(MocktailCard);
