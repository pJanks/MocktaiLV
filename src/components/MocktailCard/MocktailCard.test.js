import React from 'react';
import { MocktailCard, mapDispatchToProps, mapStateToProps } from './MocktailCard';
import { shallow } from 'enzyme';
import { addFavoriteDrink, getDrink } from '../../actions';

describe('MocktailCard', () => {
  let wrapper, mockUser, instance;

  beforeEach(() => {
    wrapper = shallow(<MocktailCard drink={ {
      strDrink: "Afterglow",
      strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/vuquyv1468876052.jpg",
      idDrink: "12560"
    } } />)
    instance = wrapper.instance()
    mockUser = {
      name: 'Greg',
      userVerified: true
    }
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the addFavoriteDrink action when addToUserFavorites is called', () => {
      const mockDispatch = jest.fn();

      const actionToDispatch = addFavoriteDrink(mockUser)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addToUserFavorites(mockUser)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch with the getDrink action when addDrinkToStore is called', () => {
      const mockDispatch = jest.fn();

      const actionToDispatch = getDrink(mockUser)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addDrinkToStore(mockUser)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

  // it('should track user input entry', () => {
  //   const mockUserEvent = { target: { name: 'user', value: 'user1' } };
  //   const mockPassEvent = { target: { name: 'password', value: 'passTest' } };
  //
  //   wrapper.instance().handleChange = jest.fn();
  //   wrapper.instance().forceUpdate();
  //   wrapper.find('[name="username"]').simulate('change', mockUserEvent);
  //   wrapper.find('[name="password"]').simulate('change', mockPassEvent);
  //
  //   expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockUserEvent);
  //   expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockPassEvent);
  // });
  //
  //
  // })
  });

  describe('mockStateToProps', () => {

    it('should return a user object', () => {
      const mockState = { user: mockUser }
      const expected = mockUser
      const mappedProps = mapStateToProps( mockState );
      expect(mappedProps.user).toEqual(expected)
    })
  })
});
