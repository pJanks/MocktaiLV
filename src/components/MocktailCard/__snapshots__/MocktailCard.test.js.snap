// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MocktailCard should match snapshot 1`] = `
<article>
  <h3>
    Afterglow
  </h3>
  <button
    onClick={[Function]}
  >
    Favorite
  </button>
  <button
    onClick={[Function]}
  >
    <Link
      className="link"
      to="/DrinkDetails12560"
    >
      Details
    </Link>
  </button>
  <img
    alt="drink image"
    className="drink-image"
    src="https://www.thecocktaildb.com/images/media/drink/vuquyv1468876052.jpg"
  />
</article>
`;
